proxy.go

```diff
@@ -43,6 +44,25 @@ func newReverseProxy() *reverseProxy {
 			// Suppress error logging in ReverseProxy, since all the errors
 			// are handled and logged in the code below.
 			ErrorLog: log.NilLogger,
+			ErrorHandler: func(rw http.ResponseWriter, req *http.Request, err error) {
+				log.Errorf("http: proxy error: %v", err)
+				rw.WriteHeader(http.StatusBadGateway)
+			},
+
+			Transport: &http.Transport {
+				DisableKeepAlives: true,
+				Proxy: http.ProxyFromEnvironment,
+				DialContext: (&net.Dialer{
+					Timeout:   30 * time.Second,
+					KeepAlive: 30 * time.Second,
+					DualStack: true,
+				}).DialContext,
+				ForceAttemptHTTP2:     true,
+				MaxIdleConns:          100,
+				IdleConnTimeout:       90 * time.Second,
+				TLSHandshakeTimeout:   10 * time.Second,
+				ExpectContinueTimeout: 1 * time.Second,
+			},
```

Comment:
Hi, thanks for your PR. 

First, how to properly check that there is no regression in chproxy bringing this feature ? Do. you see any way to properly test that ?

Also, I have the feeling that we won't have the annoying error message in the source application but way more connections made over time, one query generating one connection. This could be very impactful  on heavy loaded systems.

Couldn't we achieve the same behavior with a custom retry implemented ?

Reply:
first,this test is not very easy to test, because this problem is not easy to repeat. We tested it through pressure test.We also tried to use retry, but we couldn't guarantee idempotency.Proxy requests one connection for each connection, observe that the system is not loaded high, we can also configure to close this KeepAlive

Reply:
`observe that the system is not loaded high` is not a fact but subject to appreciation.

I'am not in favor of adding to CHProxy something that fits you, but could potentially impair a majority.

For the record, One of my production cluster is actually running between 50 an 100 QPS. The keep alive thing might have a disastrous effect on both microservices / chproxy and clickhouse.

Reply:
ok,I first submit the cannot reach chhost error log PR, which provides details about the connection CH exception, and then I will see if I can choose to close KeepAlive by adding configuration